controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -195,15 +194,15 @@ void dynamic_channel_selection_task::work()
     case eState::SCAN_DONE: {
         LOG(TRACE) << "SCAN_DONE";
 
-        if (m_scanning != id) {
+        if (m_scanning_task_id != id) {
             TASK_LOG(ERROR) << "scan done event received for a task that isn't the scanning task";
             m_last_scan_error_code = beerocks::eChannelScanErrCode::CHANNEL_SCAN_INTERNAL_FAILURE;
             fsm_move_state(eState::ABORT_SCAN);
```

Comment:
Why do we `ABORT` the scan in this case? How can this even happen?

Reply:
it shouldn't happen - but wanted to be extra safe here and not update the `m_scanning_task_id` and the results if for some reason this state is reached even though we don't own the scanning-permission